lib/timers.js

```diff
@@ -554,56 +557,66 @@ function tryOnImmediate(immediate, queue) {
   }
 }
 
+function runCallback(timer) {
+  var argv = timer._argv;
+  var argc = argv ? argv.length : 0;
+  switch (argc) {
+    // fast-path callbacks with 0-3 arguments
+    case 0:
+      return timer._callback();
+    case 1:
+      return timer._callback(argv[0]);
+    case 2:
+      return timer._callback(argv[0], argv[1]);
+    case 3:
+      return timer._callback(argv[0], argv[1], argv[2]);
+    // more then 3 arguments run slower with .apply
+    default:
+      return timer._callback.apply(timer, argv);
+  }
+}
 
-function Immediate() { }
-
-Immediate.prototype.domain = undefined;
-Immediate.prototype._onImmediate = undefined;
-Immediate.prototype._idleNext = undefined;
-Immediate.prototype._idlePrev = undefined;
 
+function createImmediate(callback) {
+  return {
+    _idleNext: null,
+    _idlePrev: null,
+    _callback: callback,
+    _argv: null,
```

Comment:
Hmm.  So passing args to the create function runs at the same speed, but moving the
create to below the args gathering and leaving the var as `const` drops the speed.
Creating the immediate as `var` immediate = createImmediate(callback, args) is 50% faster
than the almost identical `const` immediate = ... .  In that location -- above the switch
const and var run the same.  This I don't understand (maybe a v8 scoping optimization?)

Stylistically, I like not having to poke attributes into the object, but creating it up top
and populating it below may be a clearer flow, it's how setTimeout and setInterval work.